libcontainer/apparmor/apparmor.go

```diff
@@ -6,17 +6,28 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"os/exec"
 
 	"github.com/opencontainers/runc/libcontainer/utils"
 )
 
 // IsEnabled returns true if apparmor is enabled for the host.
 func IsEnabled() bool {
 	if _, err := os.Stat("/sys/kernel/security/apparmor"); err == nil && os.Getenv("container") == "" {
-		if _, err = os.Stat("/sbin/apparmor_parser"); err == nil {
+		isEnabledByParameters := func() bool {
```

Comment:
The code looks way too complicated to me, especially with an introduced function.

We can do something like this instead
```go
if _, err := os.Stat("/sys/kernel/security/apparmor"); err != nil {
  return false
}
if os.Getenv("container") != "" {
  return false
}
if _, err = os.Stat("/sbin/apparmor_parser"); err != nil {
  if _, err = exec.LookPath("apparmor_parser"); err != nil {
    return false
  }
}
buf, err := ioutil.ReadFile("/sys/module/apparmor/parameters/enabled")
return err == nil && len(buf) > 1 && buf[0] == 'Y'
```

Reply:
See my comment -- I don't think checking for `apparmor_parser` is necessary. Also why do we have the `container` env check? Does LXC set that or something?

Reply:
Dusting off git history, the check for container env var comes from https://github.com/moby/moby/pull/5534. And yes, `container=` was set at least by Docker's LXC driver.

I think it's obsoleted by now and should be removed.

Reply:
I agree -- let's drop the `container` check too.